Lib/dataclasses.py

```diff
@@ -1059,12 +1065,36 @@ def _process_class(cls, init, repr, eq, order, unsafe_hash, frozen,
         # Does this class have a post-init function?
         has_post_init = hasattr(cls, _POST_INIT_NAME)
 
+        # Does it only accept kwargs?
+        kw_only_post_init = False
```

Comment:
After rethinking, this entire loop and if could be replaced with(pseudo code)`kw_only_post_init = len(post_init_params) > 1 and post_init_params.values()[1].kind == KEYWORD_ONLY